### PR TITLE
boards: stm32h7: nucleo-h743zi2: add watchdog

### PR DESCRIPTION
**Pull request by slorquet**
## Summary
Add a config to test the stm32h7 watchdog

## Impact
None

## Testing
Builds and run

### CODE REVIEW

File: boards/arm/stm32h7/nucleo-h743zi2/src/stm32_bringup.c [hunk 1]
```diff
@@ -96,6 +96,21 @@ int stm32_bringup(void)
     }
 #endif
 
+#ifdef CONFIG_STM32_IWDG
+  /* Initialize the watchdog timer */
+
+  stm32_iwdginitialize("/dev/watchdog0", STM32_LSI_FREQUENCY);
```
Comment (xiaoxiang781216) [inline boards/arm/stm32h7/nucleo-h743zi2/src/stm32_bringup.c:107]: @slorquet you can try the built-in watchdog monitor instead:
https://github.com/apache/nuttx/blob/master/drivers/timers/Kconfig#L395-L464

**Review by xiaoxiang781216 — APPROVED** [review]

**Review by xiaoxiang781216 — APPROVED** [review]

**Review by jerpelea — COMMENTED** [review]
please add a proper commit title and commit message 
ex boards: stm32h7: nucleo: add watchdog

File: boards/arm/stm32h7/nucleo-h743zi2/src/stm32_bringup.c (hunk 1 above)
Comment (slorquet) [reply boards/arm/stm32h7/nucleo-h743zi2/src/stm32_bringup.c:107]: Why do so? I need to control which task resets the watchdog.

**Comment by slorquet** [comment]
There are two commits. how do I edit the previous commit?

File: boards/arm/stm32h7/nucleo-h743zi2/src/stm32_bringup.c (hunk 1 above)
Comment (slorquet) [reply boards/arm/stm32h7/nucleo-h743zi2/src/stm32_bringup.c:107]: Followup: I did that and that works.

However, it ENTIRELY masks why I got stack corruption at context switch when trying to create a kthread like I wanted to contribute. This is still worrying.

**Comment by patacongo** [comment]
> Contributing to nuttx is painful, specifically because of all these board tests that test canned config and miss new features.

I tend to agree.  Things are not as much fun as the old days.  But with so many committers and personal with absolute authority, I think we as a group of 50 or contributors need the discipline of strict process to keep things from degenerating into chaos.

**Comment by f4grx** [comment]
Surely there are reasonable reasons for this.

It will limit me to critical contributions I can't avoid.

**Comment by slorquet** [comment]
Surely there are reasonable reasons for this.

It will limit me to critical contributions I can't avoid. Because I dont have spoons to spend beautifying pull requests when a workaround has been found.

**Pull request closed without merging**